#!/bin/sh -e
## 13_intsign-fixes.dpatch by Henrique de Moraes Holschuh <hmh@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix signed/unsigned comparison issues

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -urNad hplip/io/hpiod/system.cpp /tmp/dpep.5IUSPn/hplip/io/hpiod/system.cpp
--- hplip/io/hpiod/system.cpp	2005-05-19 14:02:48.955043693 -0300
+++ /tmp/dpep.5IUSPn/hplip/io/hpiod/system.cpp	2005-05-19 14:11:53.858988582 -0300
@@ -711,15 +711,14 @@
    return err;
 }
 
-int System::SetSnmp(char *ip, int port, char *szoid, int type, unsigned char *buffer, int size, int *pml_result, int *result)
+int System::SetSnmp(char *ip, int port, char *szoid, int type, unsigned char *buffer, unsigned int size, int *pml_result, int *result)
 {
    struct snmp_session session, *ss=NULL;
    struct snmp_pdu *pdu=NULL;
    struct snmp_pdu *response=NULL;
    oid anOID[MAX_OID_LEN];
    size_t anOID_len = MAX_OID_LEN;
-   int len=0;
-   unsigned int i;
+   unsigned int i, len=0;
    uint32_t val;
 
    *result = R_IO_ERROR;
@@ -744,7 +743,7 @@
       case PML_DT_ENUMERATION:
       case PML_DT_SIGNED_INTEGER:
          /* Convert PML big-endian to SNMP little-endian byte stream. */
-         for(i=0, val=0; i<(unsigned int)size && i<sizeof(val); i++)    
+         for(i=0, val=0; i<size && i<sizeof(val); i++)    
             val = ((val << 8) | buffer[i]);
          snmp_pdu_add_variable(pdu, anOID, anOID_len, ASN_INTEGER, (unsigned char *)&val, sizeof(val));
          break;
@@ -779,12 +778,12 @@
    return len;
 }
 
-int System::GetSnmp(char *ip, int port, char *szoid, unsigned char *buffer, int size, int *type, int *pml_result, int *result)
+int System::GetSnmp(char *ip, int port, char *szoid, unsigned char *buffer, unsigned int size, int *type, int *pml_result, int *result)
 {
    struct snmp_session session, *ss=NULL;
    struct snmp_pdu *pdu=NULL;
    struct snmp_pdu *response=NULL;
-   int i, len=0;
+   unsigned int i, len=0;
    oid anOID[MAX_OID_LEN];
    size_t anOID_len = MAX_OID_LEN;
    struct variable_list *vars;
@@ -823,7 +822,7 @@
             *type = PML_DT_SIGNED_INTEGER;
 
             /* Convert SNMP little-endian to PML big-endian byte stream. */
-            len = (sizeof(uint32_t) < (unsigned int)size) ? sizeof(uint32_t) : size;
+            len = (sizeof(uint32_t) < size) ? sizeof(uint32_t) : size;
             val = *vars->val.integer;
             for(i=len-1; i>0; i--)
             {
@@ -843,7 +842,7 @@
             break;
          case ASN_OCTET_STR:
             *type = PML_DT_STRING;
-            len = (vars->val_len < (unsigned int)size) ? vars->val_len : size;
+            len = (vars->val_len < size) ? vars->val_len : size;
             memcpy(buffer, vars->val.string, len);
             break;
          default:
diff -urNad hplip/io/hpiod/system.h /tmp/dpep.5IUSPn/hplip/io/hpiod/system.h
--- hplip/io/hpiod/system.h	2005-05-13 14:36:42.650153286 -0300
+++ /tmp/dpep.5IUSPn/hplip/io/hpiod/system.h	2005-05-19 14:02:49.179016612 -0300
@@ -75,8 +75,8 @@
    int DeviceCleanUp(SessionAttributes *sa);
    int SetPml(int device, int channel, char *snmp_iod, int type, unsigned char *data, int dataLen, char *sendBuf);
    int GetPml(int device, int channel, char *snmp_iod, char *sendBuf);
-   int GetSnmp(char *ip, int port, char *szoid, unsigned char *buffer, int size, int *type, int *pml_result, int *result);
-   int SetSnmp(char *ip, int port, char *szoid, int type, unsigned char *buffer, int size, int *pml_result, int *result);
+   int GetSnmp(char *ip, int port, char *szoid, unsigned char *buffer, unsigned int size, int *type, int *pml_result, int *result);
+   int SetSnmp(char *ip, int port, char *szoid, int type, unsigned char *buffer, unsigned int size, int *pml_result, int *result);
    int MakeUriFromIP(char *ip, int port, char *sendBuf);
    int MakeUriFromDevice(char *dnode, char *sendBuf);
 
